### PR TITLE
mamba release 2023.02.09 (libmamba, libmambapy 1.3.1)

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Jira ticket:** [PKG-1069](https://anaconda.atlassian.net/browse/PKG-1069)

**The upstream data:**

1. **mamba** project
- Changelog: https://github.com/mamba-org/mamba/blob/2023.02.09/CHANGELOG.md#20230209
- License: https://github.com/mamba-org/mamba/blob/2023.02.09/LICENSE
- Requirements:
  - https://github.com/mamba-org/mamba/blob/2023.02.09/CMakeLists.txt
  -  https://github.com/mamba-org/mamba/blob/2023.02.09/pyproject.toml

2. **libmamba**
- Changelog: https://github.com/mamba-org/mamba/blob/2023.02.09/libmamba/CHANGELOG.md
- License: https://github.com/mamba-org/mamba/blob/2023.02.09/libmamba/LICENSE
- Requirements:
  - https://github.com/mamba-org/mamba/blob/2023.02.09/libmamba/CMakeLists.txt

3. **libmambapy**
- Changelog: https://github.com/mamba-org/mamba/blob/2023.02.09/libmambapy/CHANGELOG.md
- License:
- Requirements:
  - https://github.com/mamba-org/mamba/blob/2023.02.09/libmambapy/CMakeLists.txt
  - https://github.com/mamba-org/mamba/blob/2023.02.09/libmambapy/setup.cfg
  - https://github.com/mamba-org/mamba/blob/2023.02.09/libmambapy/setup.py

**Actions**:

- Add pinning for reproc-cpp, reproc: 14.2.4

- Add pinning for spdlog 1.11.0

- Add pinning for fmt 9.1.0

- Skip py<37 for libmambapy

- Update pinning for conda in mamba subpackage

- Add -DCMAKE_BUILD_TYPE=Release to build scritps

- Fix exits in build scripts

- Update doc_url

**Notes**:

- `libmamba 1.3.0` requires `styled` https://fmt.dev/9.1.0/api.html#_CPPv4I0EN3fmt6styledEN6detail10styled_argI14remove_cvref_tI1TEEERK1T10text_style missing in the current `fmt 8.1.1`, so we update `fmt` to **9.1.0** https://github.com/AnacondaRecipes/fmt-feedstock/pull/4.
But the current version **1.9.2** of spdlog has `fmt` with pinning `8.1.1`.
- For building `libmamba 1.3.0` conda-forge uses `fmt 9.1.0` and `spglog 1.11.0`, see https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=652819&view=logs&j=ced5d8de-8227-5f3f-33a1-45cf1592c45a&t=f7bcb349-6dca-53ba-d88d-1d9963e119ba
- packages that depend on `fmt`: 
  - `libmamba`, `libmambapy`, and `spdlog` have pinning `fmt >=8.1.1,<9.0a0`,
  - `omniscidb`, `omniscidb-common`, `omniscidbe`, and `pyomniscidbe` have pinning `fmt >=7.1.3,<8.0a0` https://metayaml-conda.fly.dev/metayaml/reverse_dependencies?name=fmt

### CODE REVIEW

File: recipe/cmake_deps.patch [hunk 1]
```diff
@@ -0,0 +1,27 @@
+diff --git a/libmamba/libmambaConfig.cmake.in b/libmamba/libmambaConfig.cmake.in
+index d5260585..7c3a2b06 100644
+--- a/libmamba/libmambaConfig.cmake.in
```
Comment (cbouss) [inline recipe/cmake_deps.patch]: Since this patch is unused, can it be removed?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -51,29 +42,29 @@ outputs:
         - nlohmann_json
         - cpp-expected
         - reproc-cpp 14.2.4
-        - spdlog
+        - reproc 14.2.4
+        - spdlog 1.11.0
         - yaml-cpp
         - termcolor-cpp
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
termcolor dropped as of 1.1.0

File: recipe/meta.yaml [hunk 3]
```diff
@@ -51,29 +42,29 @@ outputs:
...
         - cli11
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
This seems to be a test dependency.
cli11 is not listed in the cmakelist and only used in libmamba/tests/testing/lock.cpp

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,27 +1,19 @@
 {% set name = "mamba" %}
-{% set libmamba_version = "1.0.0" %}
-{% set libmambapy_version = "1.0.0" %}
-{% set mamba_version = "1.0.0" %}
-{% set release = "2022.11.01" %}
+{% set libmamba_version = "1.3.0" %}
+{% set libmambapy_version = "1.3.0" %}
+{% set mamba_version = "1.3.0" %}
+{% set release = "2023.02.03" %}
```
Comment (cbouss) [inline recipe/meta.yaml]: 1.3.1 with tag 2023.02.09 is out.
https://github.com/mamba-org/mamba/compare/2023.02.03...2023.02.09
You only need to update the version and sha256 to get it.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -51,29 +42,29 @@ outputs:
...
+        - reproc 14.2.4
+        - spdlog 1.11.0
         - yaml-cpp
         - termcolor-cpp
         - cli11
-        - fmt
-        - libiconv                               # [osx]
-        - winreg                                 # [win]
+        - fmt 9.1.0
+        - winreg             # [win]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
        - winreg             # [win]
        - zstd 1.5.2
        - bzip2 1.0
```
https://github.com/mamba-org/mamba/blob/2023.02.03/libmamba/CMakeLists.txt#L432

File: recipe/meta.yaml [hunk 6]
```diff
@@ -34,15 +26,14 @@ outputs:
...
         - $RPATH/ld64.so.1    # [s390x] Known s390x `ld64.so` issue.
     requirements:
       build:
         - {{ compiler('cxx') }}
         - cmake
         - ninja
-        - python  # [win]
+        - python            # [win]
       host:
         - libsolv 0.7.22
```
Comment (cbouss) [inline recipe/meta.yaml:38]: We could pin all the host dependencies here.
i.e.: `- openssl 1.1.1` ....

File: recipe/meta.yaml [hunk 7]
```diff
@@ -51,29 +42,29 @@ outputs:
...
         - yaml-cpp
         - termcolor-cpp
         - cli11
-        - fmt
-        - libiconv                               # [osx]
-        - winreg                                 # [win]
+        - fmt 9.1.0
+        - winreg             # [win]
       run:
         - libsolv >=0.7.22
```
Comment (cbouss) [inline recipe/meta.yaml]: If we want to be explicit, the run section should follow the run_exports.
E.g. - libsolv >=0.7.22<0.8

File: recipe/meta.yaml [hunk 8]
```diff
@@ -104,8 +96,8 @@ outputs:
         - openssl
```
Comment (cbouss) [inline recipe/meta.yaml]: All the following can be removed.
These are dependencies from libmamba, not listed in https://github.com/mamba-org/mamba/blob/2023.02.03/libmambapy/CMakeLists.txt
        - openssl
        - yaml-cpp
        - cpp-expected
        - spdlog 1.11.0
        - fmt 9.1.0
        - termcolor-cpp
        - nlohmann_json

**Review by cbouss — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 3 above)
Comment (ViridianMelody) [reply recipe/meta.yaml]: Moving `cli11` to test dependencies causes the build to fail.

File: recipe/meta.yaml (hunk 8 above)
Comment (ViridianMelody) [reply recipe/meta.yaml]: Removing any of these except for `termcolor-cpp` caused build failures. I suspect `libmambapy` pulls its dependency list from `libmamba`, but I could be wrong.

File: recipe/meta.yaml (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: This file also requires cli11 https://github.com/mamba-org/mamba/blob/a94d5ac9d6450c6058d02247fe95f3175e2cab98/mamba_package/src/main.cpp#L30. But it's for disabled subpackage `mamba`

**Comment by skupr-anaconda** [comment]
@cbouss Could you check if the pinnings are correct?

**Review by cbouss — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 9]
```diff
@@ -34,46 +26,46 @@ outputs:
...
         - reproc-cpp 14.2.4
-        - spdlog
+        - reproc 14.2.4
+        - spdlog 1.11.0
         - yaml-cpp
-        - termcolor-cpp
+        - fmt 9.1.0
+        - winreg  # [win]
+        - zstd 1.5.2
+        - bzip2 1.0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:51]: @cbouss It's not clear to me what versions should be for packages that we do not use in `run`. I add hard pinnings only for `openssl`, `reproc`, `spdlog`, `fmt`, `zstd`, and `bzip`.

Comment (cbouss) [reply recipe/meta.yaml:51]: Most have run_exports, so still end up in the run section (libcurl, ,libarchive..).
The advantage of explicit pins is for later rebuilds. That being said that does involve heavy manual work at the moment and I am not clear and what should be the proper/easy way.
I am fine with the current state of the recipe.

**Pull request merged**